src/main/java/com/wavefront/sdk/direct/ingestion/WavefrontDirectIngestionClient.java

```diff
@@ -383,6 +410,20 @@ private void internalFlush(LinkedBlockingQueue<String> buffer, String format, St
     }
   }
 
+  private void requeue(LinkedBlockingQueue<String> buffer, List<String> items,
+                       WavefrontSdkCounter dropped, String entityType) {
+    int numAddedBackToBuffer = 0;
+    for (String item : items) {
+      if (buffer.offer(item)) {
+        numAddedBackToBuffer++;
+      } else {
+        dropped.inc(items.size() - numAddedBackToBuffer);
+        logger.log(Level.WARNING, "Buffer full, dropping attempted \"" + entityType + "\"");
```

Comment:
also log up to once every 5 seconds

Reply:
maybe also print the number of entities dropped?

Reply:
done